DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc

```diff
@@ -554,12 +567,19 @@ void Converter<DDLElementaryMaterial>::operator()(xml_h element) const {
       // A is Mass of a mole in Geant4 units for atoms with atomic shell
       printout(ns.context()->debug_materials ? ALWAYS : DEBUG,
                "DD4CMS",
-               "    ROOT definition of %-50s Atomic weight %.3f, Atomic number %u, Number of nucleons %u",
+               "    ROOT definition of %-50s Atomic weight %g, Atomic number %u, Number of nucleons %u",
                elt->GetName(),
-               (elt->A()),
+               elt->A(),
                elt->Z(),
                elt->N());
-      if (atomicNumber != elt->Z() || atomicWeight != elt->A())
+      printout(ns.context()->debug_materials ? ALWAYS : DEBUG,
+               "DD4CMS",
+               "+++ Compared to XML values: Atomic weight %g, Atomic number %u",
+               atomicWeight,
+               atomicNumber);
+      static const double weightTolerance = 0.001;
```

Comment:
This weight tolerance should probably be smaller to more closely match the original behavior. I'll try to push a change later today.

Reply:
@cvuosalo I understand this PR is still work in progress, so I will wait for your update